File: Python/refer-offer/shan_chu_lian_biao_de_jie_dian_lcof.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

def deleteNode(head: ListNode, val: int) -> ListNode:
    pre = ListNode(0)
    pre.next = head
    while pre.next:
        if pre.next.val == val:
            pre.next = pre.next.next
            continue
        else:
            pre = pre.next
    return pre.next

File: core/tmp/Java/src/com/dbc/algorithm/ReferOffer/ShanChuLianBiaoDeJieDianLcof.java
package com.dbc.algorithm.ReferOffer;

public class ShanChuLianBiaoDeJieDianLcof {
    public class ListNode {
        int val;
        ListNode next;

        ListNode(int x) {
            val = x;
        }
    }

    public ListNode deleteNode(ListNode head, int val) {
        ListNode pre = new ListNode(0);
        pre.next = head;
        ListNode temp = pre;
        while (temp.next != null) {
            if (temp.next.val == val) {
                temp.next = temp.next.next;
                break;
            } else {
                temp = temp.next;
            }
        }
        return pre.next;
    }
}
